tinygrad/runtime/ops_clang.py

```diff
@@ -1,24 +1,37 @@
-import ctypes, subprocess, pathlib, tempfile
+import ctypes, subprocess, pathlib, tempfile, mmap, platform
 from tinygrad.device import Compiled, Compiler, MallocAllocator
-from tinygrad.helpers import cpu_time_execution, DEBUG, cpu_objdump
+from tinygrad.helpers import OSX, cpu_time_execution, DEBUG, cpu_objdump, mv_address
 from tinygrad.renderer.cstyle import ClangRenderer
+from tinygrad.runtime.support.elf import elf_loader
+from tinygrad.runtime.autogen import libc
 
 class ClangCompiler(Compiler):
   def compile(self, src:str) -> bytes:
+    triple = f"{platform.machine()}-none-unknown-elf"
+    asm = subprocess.check_output(['clang', '-march=native', '-O2', '-Wall', '-Werror', '-x', 'c', '-S', '-ffreestanding', '-nostdlib', '-target',
```

Comment:
There has to be a better way than this. subprocess is really expensive

Reply:
Will look at using `libclang` instead, with autogenerated bindings.

Reply:
So from what I can tell, `libclang` doesn't actually support compilation, and is really just made for AST browsing. `clang -cc1` (the compiler frontend) is actually implemented using `libclang-cpp`, which makes no guarantees about a stable API, and, as the name suggests, is written in C++, which would make calling out to it from Python much harder (not to mention we'd likely have to require a specific version of Clang and LLVM to be installed because of the unstable API). There are a couple projects I found that also use this API, such as [DragonFFI](https://github.com/aguinet/dragonffi), [Cling](https://github.com/root-project/cling), and [this blog post](https://blog.memzero.de/libclang-c-to-llvm-ir/), but all these are written in C++.